OpenRA.Game/SelectableExts.cs

```diff
@@ -73,14 +73,15 @@ public static FrozenActor WithHighestSelectionPriority(this IEnumerable<FrozenAc
 			return actors.MaxByOrDefault(a => CalculateActorSelectionPriority(a.Info, a.MouseBounds, selectionPixel, modifiers));
 		}
 
-		static long CalculateActorSelectionPriority(ActorInfo info, Rectangle bounds, int2 selectionPixel, Modifiers modifiers)
+		static long CalculateActorSelectionPriority(ActorInfo info, Polygon bounds, int2 selectionPixel, Modifiers modifiers)
 		{
 			if (bounds.IsEmpty)
 				return info.SelectionPriority(modifiers);
 
+			var br = bounds.BoundingRect;
 			var centerPixel = new int2(
-				bounds.Left + bounds.Size.Width / 2,
-				bounds.Top + bounds.Size.Height / 2);
+				br.Left + br.Size.Width / 2,
+				br.Top + br.Size.Height / 2);
```

Comment:
Just noting that the centroid need not be the same as the bounding box center. An example is a regular pentagon, which is also an example of radial symmetry since all regular polygons have rotational symmetry. Polygons with rotational symmetry of a degree divisible by four and polygons with both horizontal and vertical symmetry such as the isometric and dimetric projections of a cube do have the center of their bounding boxes as their centroids. The TS mod w is an example. Modders should avoid shapes that deviate too much from those two sets since that would cause irregularities in selection edge cases.